### PR TITLE
Api module release notes support to validate

### PR DESCRIPTION
**Pull request by DeanArbel**
## Status
- [x] Ready

## Related Issues
fixes: https://github.com/demisto/etc/issues/24157
continues PR: https://github.com/demisto/demisto-sdk/pull/744

## Description
Add the following behavior:
* demisto-sdk validate validates that the APIModule change is accompanied by release notes for all related packs.

## Screenshots
Paste here any images that will help the reviewer
image

## Must have
- [x] Tests

### CODE REVIEW

File: demisto_sdk/commands/validate/validate_manager.py [hunk 1]
```diff
@@ -692,6 +697,15 @@ def validate_no_missing_release_notes(self, modified_files, old_format_files, ad
                                                                                    FileType.TEST_PLAYBOOK,
                                                                                    FileType.TEST_SCRIPT,
                                                                                    FileType.DOC_IMAGE})
+        if 'ApiModules' in packs_that_should_have_new_rn:
```
Comment (bakatzir) [inline demisto_sdk/commands/validate/validate_manager.py:700]:
```suggestion
        if API_MODULES_PACK in packs_that_should_have_new_rn:
```

File: demisto_sdk/commands/validate/validate_manager.py [hunk 2]
```diff
@@ -692,6 +697,15 @@ def validate_no_missing_release_notes(self, modified_files, old_format_files, ad
...
+            if not os.path.isfile(self.id_set_path):
+                IDSetCreator(print_logs=False, output=self.id_set_path).create_id_set()
```
Comment (bakatzir) [inline demisto_sdk/commands/validate/validate_manager.py:702]: didn't you create it in 623? why again or am I missing something?

File: demisto_sdk/commands/validate/tests/validators_test.py [hunk 3]
```diff
@@ -635,6 +636,44 @@ def test_validate_no_missing_release_notes__missing_rn_in_modified_files(self, r
                                                                       old_format_files=set(),
                                                                       added_files=added_files) is False
 
+    def test_validate_no_missing_release_notes__missing_rn_dependent_on_api_module(self, repo, mocker, tmpdir):
```
Comment (bakatzir) [inline demisto_sdk/commands/validate/tests/validators_test.py:639]: care to add another test which is positive(all packs depending on the apimodules have release notes as well)?

File: demisto_sdk/commands/validate/validate_manager.py (hunk 2 above)
Comment (DeanArbel) [reply demisto_sdk/commands/validate/validate_manager.py:702]: technically yes, but actually no. 
line 623 is for `validate_changed_packs_unique_files`
line 702 is for `validate_no_missing_release_notes`

so no, but they are both called from `run_validation_using_git` - which calls 623 before 702, so technically yes

**Review by bakatzir — APPROVED** [review]

**Pull request merged**